Docs/Benchmarks.md

```diff
@@ -30,16 +35,59 @@ Here you can find some measurements of execution times of various algorithms on
 
 ### Vitis
 
-Comparing the performance of a Vitis platform FPGA (Xilinx Alveo U280) to the host PC's performance on a [Nimbix](https://www.nimbix.net/alveo) "Xilinx Vitis Unified Software Platform 2019.2" instance. Only a single CPU is assumed to be running under 100% load for the power usage figures for the sake of simplicity.
+Comparing the performance of a Vitis platform FPGA (Xilinx Alveo U280) to the host PC's performance on a [Nimbix](https://www.nimbix.net/alveo) "Xilinx Vitis Unified Software Platform 2020.1" instance. Only a single CPU is assumed to be running under 100% load for the power usage figures for the sake of simplicity. The table has a matching [Excel sheet](BenchmarksVitis.xlsx) that was converted using [this VS Code extension](https://marketplace.visualstudio.com/items?itemName=csholmq.excel-to-markdown-table).

```

Comment:
Ez továbbra is érvényes nem?

> Mindig a második futást nézzük, azaz amikor a PCIe setup és a .NET JIT megvolt. Ezt írjuk is le a benchmarkokhoz.

Reply:
Nem, mert a harmadikat. xD 

Kicsit lejjebb a snippet mutatja, hogy kétszer lefutott eldobott értékekkel és a harmadikat írta ki.
```bash
	dotnet Hast.Samples.Consumer.dll -sample $sample -device "$device" -name "$name" > /dev/null
	dotnet Hast.Samples.Consumer.dll -sample $sample -device "$device" -name "$name" > /dev/null
	dotnet Hast.Samples.Consumer.dll -sample $sample -device "$device" -name "$name"
```

Most hozzáírom még:
"- Measurements were always taken from the second or third run of the application to disregard initialization time."

Reply:
Ja OK, az is jó.

Reply:
A PCIe setupra amúgy ez miért megoldás? A JIT-re OK, de itt ugye három különböző alkalomkor fut le az egész app, szóval akármit is tölt be, az közte elveszik. Az appon belül kellene többször futtatni.

Reply:
Attól függ mit értünk PCIe setup alatt. Ez felébreszti a kártyát, ha kell.

A kernel feltöltése viszont más kategória. A communication service minden híváskor létre hozza a kernelt így az appon belüli többszörös futtatás nem érne el semmit. Ez azért van, mert program elindításához (amikor feltöltöd a command queue-ba) el kell küldeni a kernelt is. Ez normális OpenCL pattern. Azt elvileg a xilinx firmware adja, hogy ha az épp betöltött kernel érkezik újra, akkor ne programozza újra a kártyát. De amúgy ha van is emiatt különbség, érzékelhetetlenül kicsi. Amikor futtattam az ImageContrastModifier-t a nagy képpel tízszer egymás után akkor láttam ±0.02% változást, de random bármilyen trend nélkül.

Reply:
Hmm, hol van pontosan ez a kernel feltöltés?

Reply:
`OpenClCommunicationService.Execute`-ban a `_binaryOpenCl.CreateBinaryKernel(kernelBinary, KernelName);` hozza létre a natív objektumot [itt](https://github.com/Lombiq/Hastlayer-SDK/blob/dev/Hast.Abstractions/Hast.Vitis.Abstractions/Services/OpenClCommunicationService.cs#L84). Abban nem vagyok 100% biztos, hogy már ilyenkor elküldi az eszköznek vagy csak kicsit lejjebb a program elindításakor (`_binaryOpenCl.LaunchKernel(deviceIndex, KernelName, new[] { fpgaBuffer });`) [itt](https://github.com/Lombiq/Hastlayer-SDK/blob/dev/Hast.Abstractions/Hast.Vitis.Abstractions/Services/OpenClCommunicationService.cs#L132). Azért nem egyértelmű, mert a kernel feltöltését az OpenCL intézi.

Reply:
Aha, köszi. Szóval akkor ez az elküldés nem spórolható meg és OpenCL-ben mindenki mindenhol azt csinálja, hogy minden hívás előtt újra elküldi?

Reply:
Legalább is én úgy tudom.
Aztán lehet itt is van valami alap dolog amiről nem tudunk, mint ez a 2019.2 vs 2020.1 build ami most derült ki. :/

Reply:
OK, egyelőre hagyjuk.